pandas/_libs/algos.pyx

```diff
@@ -96,22 +96,94 @@ class NegInfinity(object):
     __ge__ = lambda self, other: self is other
 
 
-cdef inline Py_ssize_t swap(numeric *a, numeric *b) nogil except -1:
-    cdef numeric t
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def is_lexsorted(list list_of_arrays):
+    cdef:
+        int i
+        Py_ssize_t n, nlevels
+        int64_t k, cur, pre
+        ndarray arr
+        bint result = 1
+
+    nlevels = len(list_of_arrays)
+    n = len(list_of_arrays[0])
+
+    cdef int64_t **vecs = <int64_t**> malloc(nlevels * sizeof(int64_t*))
+    for i from 0 <= i < nlevels:
```

Comment:
Since moving all this could take the opportunity to update loops using this old syntax to `range(...)`, though not a big deal